src/include/duckdb/common/adbc/adbc.hpp

```diff
@@ -10,10 +10,255 @@
 
 #include "duckdb/common/adbc/adbc.h"
 
+#include "duckdb/main/capi/capi_internal.hpp"
+
 #include <string>
 
 namespace duckdb_adbc {
 
+// RAII wrappers for C-API resources used in Ingest
+class ArrowSchemaWrapper {
+public:
+	ArrowSchemaWrapper() : schema(new ArrowSchema) {
+	}
+	~ArrowSchemaWrapper() {
+		if (schema) {
+			if (schema->release) {
+				schema->release(schema);
+			}
+			delete schema;
+		}
+	}
+	ArrowSchema *get() const {
+		return schema;
+	}
+	ArrowSchema &operator*() const {
+		return *schema;
+	}
+	ArrowSchema *operator->() const {
+		return schema;
+	}
+	ArrowSchemaWrapper(const ArrowSchemaWrapper &) = delete;
+	ArrowSchemaWrapper &operator=(const ArrowSchemaWrapper &) = delete;
+	ArrowSchemaWrapper(ArrowSchemaWrapper &&other) noexcept : schema(other.schema) {
+		other.schema = nullptr;
+	}
+	ArrowSchemaWrapper &operator=(ArrowSchemaWrapper &&other) noexcept {
+		if (this != &other) {
+			delete schema;
```

Comment:
This is wrong, the `schema` could be holding memory that first needs to be `release()`'d